jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -631,6 +634,7 @@ private static void ensureProperties(boolean isInitialization) {
 		}
 		initializedProperties.put(key, list[i+1]);
 	}
+	initializedProperties.put("file.encoding", fileEncoding); //$NON-NLS-1$
```

Comment:
Similarly I don't think this change is needed along with the system.c change.

Reply:
This is still needed to prevent from overwriting by command line options via `initializedProperties.put(key, list[i+1])`.

Reply:
Isn't fileEncoding set to the command line option? I think this line is only needed for when file.encoding isn't set on the command line.

Reply:
`-Dfile.encoding=COMPAT` is required to be mapped to platform encoding as per JEP 400. When it isn't set, existing code already did such mapping which is correct.

Reply:
I still don't see why this line needs to be moved. Although I don't think it's obviously harmful to do so.

Reply:
W/o this move, `file.encoding` will be set to `COMPAT` when `-Dfile.encoding=COMPAT` is specified at command line option. The native `getPropertyList()` returns the command line option settings w/o mapping like `getSysPropBeforePropertiesInitialized()`.